e2e-tests/IIoTPlatform-E2E-Tests/Orchestrated/T_TwinBrowseOrchestratedTestTheory.cs

```diff
@@ -0,0 +1,175 @@
+            }
+
+            TestHelper.Registry_ActivateEndpointAsync(_context, cts.Token).GetAwaiter().GetResult();
+        }
+
+        [Fact, PriorityOrder(2)]
+        public void D1_Discover_OPC_UA_Endpoints() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var endpoints = TestHelper.Registry_GetEndpointsAsync(_context, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotEmpty(endpoints);
+
+            var endpoint = endpoints.SingleOrDefault(e => string.Equals(_context.OpcUaEndpointId, e.Id));
+
+            Assert.False(endpoint.Id == null, "The endpoint was not found");
+            Assert.Equal(TestConstants.StateConstants.ActivatedAndConnected, endpoint.ActivationState);
+            Assert.Equal(TestConstants.StateConstants.Ready, endpoint.EndpointState);
+        }
+
+        [Fact, PriorityOrder(3)]
+        public void T1_Browse_Address_Space() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var nodes = TestHelper.Twin_GetBrowseEndpointAsync(_context, _context.OpcUaEndpointId, null, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+        }
+
+        [Fact, PriorityOrder(4)]
+        public void T1_1_BrowseSpecificNode() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var nodes = TestHelper.Twin_GetBrowseEndpointAsync(_context, _context.OpcUaEndpointId, null, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+
+            var nodeId = nodes.First().NodeId;
+
+            nodes = TestHelper.Twin_GetBrowseEndpointAsync(_context, _context.OpcUaEndpointId, nodeId).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+        }
+
+        [Fact, PriorityOrder(5)]
+        public void T1_2_BrowseAllNodesOfTypeObject() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var nodes = TestHelper.Twin_GetBrowseEndpoint_RecursiveAsync(_context, _context.OpcUaEndpointId, "Object", null, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+
+            Assert.True(nodes.Count > 150);
+        }
+
+        [Fact, PriorityOrder(6)]
+        public void T1_3_BrowseAllNodesOfTypeVariable() {
+            var cts = new CancellationTokenSource(TestConstants.MaxTestTimeoutMilliseconds);
+
+            var nodes = TestHelper.Twin_GetBrowseEndpoint_RecursiveAsync(_context, _context.OpcUaEndpointId, "Variable", null, cts.Token).GetAwaiter().GetResult();
+
+            Assert.NotNull(nodes);
+            Assert.NotEmpty(nodes);
+
+            Assert.True(nodes.Count > 150);
```

Comment:
I find this test weak. We should at least pick-up some node Id samples from the graph and and test against the results.